src/System.Console/src/System/IO/StdInStreamReader.cs

```diff
@@ -69,34 +94,52 @@ public override string ReadLine()
                 {
                     readLineStr = _readLineSB.ToString();
                     _readLineSB.Clear();
+                    Console.WriteLine();
                     return readLineStr;
                 }
-
-                else if (keyInfo.Key == ConsoleKey.Backspace)
+                else if (keyInfo.Key == ConsoleKey.Backspace || keyInfo.Key == ConsoleKey.Delete)
                 {
                     int len = _readLineSB.Length;
                     if (len > 0)
                     {
                         _readLineSB.Length = len - 1;
+
+                        if (s_moveLeftString == null)
+                        {
+                            string moveLeft = ConsolePal.TerminalBasicInfo.Instance.CursorLeftFormat;
+                            s_moveLeftString = moveLeft != null ? moveLeft + " " + moveLeft : string.Empty;
+                        }
+                        Console.Write(s_moveLeftString);
                     }
                 }
+                else if (keyInfo.Key == ConsoleKey.Tab)
+                {
+                    _readLineSB.Append(keyInfo.KeyChar);
+                    Console.Write(' ');
```

Comment:
Because after writing a \t, if you try to backspace I don't then know how far back to move the cursor. I can investigate another solution separately.